db/migrate/20150612135546_create_host_status.rb

```diff
@@ -11,7 +11,24 @@ def up
     add_column :hosts, :global_status, :integer, :default => 0, :null => false
 
     Host.includes(:host_statuses, :last_report_object).find_each do |host|
-      host.refresh_statuses
+      HostStatus.status_registry.each do |status_class|
+        status = host.get_status(status_class)
+        begin
+          status.refresh! if status.relevant?
+        rescue => e
+          # if the status is not ready to be saved because of missing migration
+          # we skip it and let it refresh by itself
+          logger.warn "skipping sub-status #{status.class} refresh, it's not ready"
+        end
+      end
+      host.host_statuses.reload
+      host.refresh_global_status
+
+      # validations could trigger dhcp/tftp orchestration, if smart proxy is unavailable
+      # migration would fail
+      unless host.save(:validate => false)
```

Comment:
I totally agree, however I believe, that the commit added the "disable_validation" attribute you are thinking about. However, it's called "importing_facts". @dLobatog liked that better ;-)
So you could do something like:

```ruby
host.importing_facts = true
host.save
ensure
host.importing_facts = false
```
What do you think?

I agree that foreman-tasks is probably the better approach instead of the callbacks. Don't worry about the ticket. It can't hurt though.